### PR TITLE
Python optimizations

### PR DESCRIPTION
**Pull request by chunheguo**
## Contributing requirements

* [x] I read the contribution guidelines in CONTRIBUTING.md.
* [x] I placed my solution in the correct solution folder.
* [x] I added a README.md with the right badge(s).
* [x] I added a Dockerfile that builds and runs my solution.
* [x] I selected `drag-race` as the target branch.
* [x] All code herein is licensed compatible with BSD-3.

```
$ python3 PrimePY.py 
Passes: 4327, Time: 10.001078526999999, Avg: 0.0023113192805639007, Limit: 1000000, Count: 78498, Valid: True

ssovest; 4327;10.001078526999999;1;algorithm=base,faithful=yes,bits=8
```

purely python improvements without using numpy or numba/jit

### CODE REVIEW

**Comment by rbergen** [comment]
I'm wondering what is the nature of the improvements you intend to make? Phrased differently, what goal(s) are you aiming for?

**Comment by chunheguo** [comment]
just using plain std python to improve the performance, without using numpy or jit.
While loops are slower than for loops, build in map functions are usually faster then for loops

File: PrimePython/solution_2/PrimePY.py [hunk 1]
```diff
@@ -33,38 +32,40 @@ def validate_results(self):                      # Check to see if this is an up
...
 
-        factor = 1
         # sqrt doesn't seem to make any difference in CPython,
         # but works much faster than "x**.5" in Pypy
         q = sqrt(self._size) / 2
         bitslen = len(self._bits)
+        q_int = ceil(q)
 
-        while factor <= q:
+        for factor in range(1, q_int):
```
Comment (marghidanu) [inline PrimePython/solution_2/PrimePY.py]: I believe the while loop might be faster than the for loop being used here.

**Comment by marghidanu** [comment]
Can you provide any numbers that show some improvements between the original code and your changes?

**Comment by chunheguo** [comment]
> Can you provide any numbers that show some improvements between the original code and your changes?

https://github.com/PlummersSoftwareLLC/Primes/pull/278/files#diff-ce75757ace6ff01ca0eeb44df7ceac668a8a04d8cce5391919d55940dcbb0f01R46

I was going off the values in the readme: 
https://github.com/PlummersSoftwareLLC/Primes/pull/278/files#diff-ce75757ace6ff01ca0eeb44df7ceac668a8a04d8cce5391919d55940dcbb0f01L57

**Comment by rbergen** [comment]
I don't/no longer see a performance-improving change in the drag-race relevant part of the solution. As such, I think we should close this.

**Pull request closed without merging**